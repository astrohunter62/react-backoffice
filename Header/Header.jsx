import React, { Children } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { withStyles } from 'material-ui/styles'
import AppBar from 'material-ui/AppBar'
import Toolbar from 'material-ui/Toolbar'
import Typography from 'material-ui/Typography'
import IconButton from 'material-ui/IconButton'
import MenuIcon from 'material-ui-icons/Menu'

const drawerWidth = 280

const styles = (theme) => ({
  root: {
    width: '100%',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: drawerWidth,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  hide: {
    display: 'none',
  },
  flex: {
    flex: 1,
  },
  title: {
    cursor: 'pointer',
    '&:hover': {
      textDecoration: 'underline',
    },
  }
})

const Header = ({
  title,
  open,
  fixed,
  handleDrawerOpen,
  onClick,
  children,
  classes
}) => (
  <div className={classes.root}>
    <AppBar
      className={classNames(classes.appBar, {
        [classes.appBarShift]: open,
      })}
      position={fixed ? 'fixed' : 'static'}
    >
      <Toolbar>
        <IconButton
          color="contrast"
          aria-label="open drawer"
          onClick={handleDrawerOpen}
          className={classNames(classes.menuButton, open && classes.hide)}
        >
          <MenuIcon />
        </IconButton>

        <Typography
          type="title"
          color="inherit"
          className={`${classes.flex} ${classes.title}`}
          onClick={onClick}
        >
          {title}
        </Typography>

        {children}
      </Toolbar>
    </AppBar>
  </div>
)

Header.propTypes = {
  title: PropTypes.string.isRequired,
  fixed: PropTypes.bool.isRequired,
  handleDrawerOpen: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Header)
