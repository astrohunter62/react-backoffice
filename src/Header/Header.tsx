import React, { FunctionComponent } from "react";
import classNames from "classnames";
import {
  AppBar,
  IconButton,
  makeStyles,
  Theme,
  Toolbar,
  Typography,
} from "@material-ui/core";
import MenuIcon from "@material-ui/icons/Menu";
import { DRAWER_WIDTH } from "../Drawer/Drawer";

const useStyles = makeStyles((theme: Theme) => ({
  appBar: {
    transition: theme.transitions.create(["margin", "width"], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    transform: "translate(0, 0)",
  },

  appBarShift: {
    width: `calc(100% - ${DRAWER_WIDTH}px)`,
    transition: theme.transitions.create(["margin", "width"], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: DRAWER_WIDTH,
  },

  isStatic: {
    position: "relative",
  },

  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  hide: {
    display: "none",
  },
  flex: {
    flex: 1,
  },

  title: {
    paddingLeft: theme.spacing(2),
    cursor: "pointer",
    "&:hover": {
      textDecoration: "underline",
    },
  },
}));

type HeaderProps = {
  isOpen?: boolean;
  title: string;
  isFixed?: boolean;
  onDrawerOpen: (...args: any[]) => any;
  onClick: (...args: any[]) => any;
};

const Header: FunctionComponent<HeaderProps> = ({
  title,
  isOpen = false,
  isFixed = false,
  onDrawerOpen,
  onClick,
  children,
}) => {
  const classes = useStyles();

  return (
    <AppBar
      className={classNames(classes.appBar, {
        [classes.appBarShift]: isOpen,
        [classes.isStatic]: !isFixed,
      })}
      position={isFixed ? "fixed" : "static"}
    >
      <Toolbar>
        <IconButton
          aria-label="open drawer"
          color="inherit"
          onClick={onDrawerOpen}
          className={classNames(classes.menuButton, isOpen && classes.hide)}
        >
          <MenuIcon />
        </IconButton>

        <Typography
          variant="h6"
          color="inherit"
          className={classNames(classes.flex, classes.title)}
          onClick={onClick}
        >
          {title}
        </Typography>

        {children}
      </Toolbar>
    </AppBar>
  );
};

export default Header;
