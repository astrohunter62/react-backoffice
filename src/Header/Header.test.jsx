import React from 'react'
import renderer from 'react-test-renderer'

import Header from './'

it('renders correctly', () => {
  const tree = renderer
    .create(<Header
      isOpen
      title="Header"
      isFixed
      isCookieInfoOpen
      onDrawerOpen={() => {}}
      onClick={() => { }}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with closed', () => {
  const tree = renderer
    .create(<Header
      title="Header"
      onDrawerOpen={() => { }}
      onClick={() => { }}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
