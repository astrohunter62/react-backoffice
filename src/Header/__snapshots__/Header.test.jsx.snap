// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-17 MuiPaper-elevation4-23 MuiAppBar-root-9 MuiAppBar-positionFixed-10 MuiAppBar-colorPrimary-15 mui-fixed Header-appBar-2 Header-appBarShift-4 Header-appBarWithCookieInfo-3"
  >
    <div
      className="MuiToolbar-root-44 MuiToolbar-gutters-45"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root-52 MuiIconButton-root-46 MuiIconButton-colorInherit-47 Header-menuButton-5 Header-hide-6"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-51"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-55"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-61"
        />
      </button>
      <h2
        className="MuiTypography-root-68 MuiTypography-title-74 MuiTypography-colorInherit-87 Header-flex-7 Header-title-8"
        onClick={[Function]}
      >
        Header
      </h2>
    </div>
  </header>
</div>
`;

exports[`renders correctly with closed 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-17 MuiPaper-elevation4-23 MuiAppBar-root-9 MuiAppBar-positionStatic-13 MuiAppBar-colorPrimary-15 Header-appBar-2"
  >
    <div
      className="MuiToolbar-root-44 MuiToolbar-gutters-45"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root-52 MuiIconButton-root-46 MuiIconButton-colorInherit-47 Header-menuButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-51"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-55"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-61"
        />
      </button>
      <h2
        className="MuiTypography-root-68 MuiTypography-title-74 MuiTypography-colorInherit-87 Header-flex-7 Header-title-8"
        onClick={[Function]}
      >
        Header
      </h2>
    </div>
  </header>
</div>
`;
